src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.Unix.cs

```diff
@@ -271,10 +256,39 @@ private static Interop.Sys.OpenFlags PreOpenConfigurationFromOptions(FileMode mo
             return flags;
         }
 
-        private void Init(string path, FileMode mode, FileAccess access, FileShare share, FileOptions options, long preallocationSize)
+        private bool TryInit(string path, FileMode mode, FileAccess access, FileShare share, FileOptions options, long preallocationSize)
```

Comment:
Usually a `Try` method does not throw. Maybe use a different prefix to avoid confusion?

Reply:
I've removed the prefix.